File: Java/Task315.java
import java.util.Scanner;

public class Task315 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        // в оголошенні можна створити декілька масивів
        double P[] = new double[17], Q[] = new double[17];
        double[] S = new double[17];// але зазвичай, окремо
        // оголошення double[] S; та double S[]; тотожні
        System.out.println("Введіть елементи масиву P");
        for (int i = 0; i < 17; i++) {
            P[i] = scanner.nextDouble();
        }
        for (int i = 0; i < 17; i++) {
            Q[i] = Math.random() * 1400 / 100 - 7;
        }
        double s = 0;
        int count = 0;
        for (double d : Q) // Цикл «for each» за елементами
            if (d < 0) { // масиву переглядає усі елемен-
                s += d; // ти масиву Q, поміщаючи їх
                count++; // по черзі у змінну d
            }
        double a = s / count;
        for (int i = 0; i < 17; i++) {
            S[i] = P[i] + a;
        }
        double p5 = P[0];
        System.out.println("Масив P:");
        for (int i = 0; i < 17; i++) {
            System.out.printf(" %8.2f", P[i]);
            if (Math.abs(P[i] - 5) < Math.abs(p5 - 5))
                p5 = P[i];
        }
        System.out.println();// перехід на нову стрічку
        System.out.printf("Найближче до 5: %10.2f \n", p5);
        double q5 = Q[0];
        System.out.println("Масив Q:");
        for (int i = 0; i < 17; i++) {
            System.out.printf(" %8.2f", Q[i]);
            if (Math.abs(Q[i] - 5) < Math.abs(q5 - 5))
                q5 = Q[i];
        }
        System.out.println();// перехід на нову стрічку
        System.out.printf("Найближче до 5 %10.2f \n", q5);
        double s5 = S[0];
        System.out.println("Масив S:");
        for (double x : S) {
            System.out.printf(" %8.2f", x);
            if (Math.abs(x - 5) < Math.abs(s5 - 5))
                s5 = x;
        }
        System.out.println();// перехід на нову стрічку
        System.out.printf("Найближче до 5: %10.2f \n", s5);
        scanner.close();
    }
}

File: Python/Task315.py
import random # модуль для випадкових чисел
P = []
Q = []
S = []
print('Введіть елементи масиву P')
for i in range(0,17):
    P.append(float(input()))
for i in range(0,17):
    Q.append(round(random.uniform(-7.0,7.0),2))
m = 0.0
k = 0
# Цикл «for each» на Python виглядає так:
for d in Q: #шукаємо середнє
    if d<0: #лише від'ємних елементів
        m+=d
        k+=1
a = m/k
for i in range(0,17):
    S.append(P[i]+a)
#Вибираємо перший елемент за найближчий до 5
P5 = P[0]
print('Масив P')
# цикл «for each» переглядає усі елементи масиву P,
# поміщаючи їх по черзі у змінну d
for d in P: 
    print(" %8.2f" % d, end=' ')
    if(abs(d - 5) < abs(P5 - 5)):
        P5 = d
print() #Перехід на новий рядок
print('Найближче до 5 число %8.2f' % P5)

Q5 = Q[0]
print('Масив Q')
for d in Q:
    print(" %8.2f" % d, end=' ')
    if(abs(d - 5) < abs(Q5 - 5)):
        Q5 = d
print() #Перехід на новий рядок
print('Найближче до 5 число %8.2f' % Q5)

S5 = S[0]
print('Масив S')
for d in S:
    print(" %8.2f" % d, end=' ')
    if(abs(d - 5) < abs(S5 - 5)):
        S5 = d
print() #Перехід на новий рядок
print('Найближче до 5 число %8.2f' % S5)
